dotCMS/src/main/java/com/dotcms/rest/CMSConfigResource.java

```diff
@@ -497,4 +500,30 @@ public Response deleteEndpoint ( @Context HttpServletRequest request,
         return responseResource.response( responseMessage.toString() );
     }
 
+    @POST
+    @Path ("/regenerateKey")
+    @Produces (MediaType.APPLICATION_JSON)
+    @Consumes (MediaType.APPLICATION_FORM_URLENCODED)
+    public Response regenerateKey ( @Context HttpServletRequest request,
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Parameter 'request' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5677177)